src/transformers/generation_flax_utils.py

```diff
@@ -326,6 +326,13 @@ def generate(
         if decoder_start_token_id is None and self.config.is_encoder_decoder:
             raise ValueError("`decoder_start_token_id` has to be defined for encoder-decoder generation.")
 
+        # decoder-only models should use left-padding for generation
+        if not self.config.is_encoder_decoder and jnp.sum(input_ids[:, -1] == pad_token_id) > 0:
+            logger.warning(
+                "A decoder-only architecture is being used, but right-padding was detected! For correct generation "
+                "results, please set `padding_side='left'` when initializing the tokenizer."
+            )
```

Comment:
I'm afraid this warning gets drown in the other logging/output the user might have.

Have we explored colored warnings for breaking behavior? (e.g. yellow)